rpcserver.go

```diff
@@ -809,15 +814,151 @@ func (r *rpcServer) VerifyProof(ctx context.Context,
 	)
 	valid := err == nil
 
-	// TODO(roasbeef): also show additional final resting anchor
-	// information, etc?
+	rpcProof, err := marshalProofFile(proofFile, -1)
+	if err != nil {
+		return nil, fmt.Errorf("unable to marshal proof: %w", err)
+	}
 
-	// TODO(roasbeef): show the final resting place of the asset?
-	return &tarorpc.ProofVerifyResponse{
-		Valid: valid,
+	return &tarorpc.VerifyProofResponse{
+		Valid:        valid,
+		DecodedProof: rpcProof,
 	}, nil
 }
 
+// DecodeProof attempts to decode a given proof file that claims to be anchored
+// at the specified genesis point.
+func (r *rpcServer) DecodeProof(ctx context.Context,
+	in *tarorpc.DecodeProofRequest) (*tarorpc.DecodeProofResponse, error) {
+
+	if len(in.RawProof) == 0 {
+		return nil, fmt.Errorf("proof file must be specified")
+	}
+
+	var proofFile proof.File
+	err := proofFile.Decode(bytes.NewReader(in.RawProof))
+	if err != nil {
+		return nil, fmt.Errorf("unable to decode proof file: %w", err)
+	}
+
+	resp, err := marshalProofFile(proofFile, in.ProofIndex)
+	if err != nil {
+		return nil, fmt.Errorf("unable to marshal proof: %w", err)
+	}
+
+	return resp, nil
+}
+
+// marshalProofFile turns a proof file into an RPC DecodeProofResponse.
+func marshalProofFile(proofFile proof.File, index int64) (*tarorpc.DecodeProofResponse, error) {
+	var (
+		decodedProof *proof.Proof
+		err          error
+	)
+
+	if index < 0 {
```

Comment:
Ok found the answer here. Alternatively we can use a `oneof` to say something like `PartialVerify` (which would accept an index) or `FullVerify`  (do the entire thing).

Reply:
for "partially verified"  we could do it in a reverse way

```
if len(proofs) >= index {
    // return err
}

// Proofs are store from older to newer.
proofIdx := proofFile.NumProofs() - 1 - idx

decodedProof, err = proofFile.ProofAt(proofIdx)
if err != nil {
	return nil, err
}
```

In that case you get by default the last proof, that is what you will usually want.

Reply:
Original intention was to only decode and return a single proof, defaulting to the latest proof if index was unspecified.

Didn't think there would be much reason to return the entire proof chain